packages/contracts/src/manager/facets/ManagerFacet.sol

```diff
@@ -0,0 +1,251 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.3;
+
+import {Modifiers} from "../libraries/LibAppStorage.sol";
+import {AccessControlStorage} from "../libraries/AccessControlStorage.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../../dollar/interfaces/IUbiquityDollarToken.sol";
+import "../../dollar/interfaces/ICurveFactory.sol";
+import "../../dollar/interfaces/IMetaPool.sol";
+import "../../dollar/core/TWAPOracleDollar3pool.sol";
+
+contract ManagerFacet is Modifiers {
+    // TODO Add a generic setter for extra addresses that needs to be linked
+    function setTwapOracleAddress(address _twapOracleAddress)
+        external
+        onlyAdmin
+    {
+        s.twapOracleAddress = _twapOracleAddress;
+        // to be removed
+
+        TWAPOracleDollar3pool oracle =
+            TWAPOracleDollar3pool(s.twapOracleAddress);
+        oracle.update();
+    }
+
+    function setDollarTokenAddress(address _dollarTokenAddress)
+        external
+        onlyAdmin
+    {
+        s.dollarTokenAddress = _dollarTokenAddress;
+    }
+
+    function setCreditTokenAddress(address _creditTokenAddress)
+        external
+        onlyAdmin
+    {
+        s.creditTokenAddress = _creditTokenAddress;
+    }
+
+    function setCreditNFTAddress(address _creditNFTAddress)
+        external
+        onlyAdmin
+    {
+        s.creditNFTAddress = _creditNFTAddress;
+    }
+
+    function setGovernanceTokenAddress(address _governanceTokenAddress)
+        external
+        onlyAdmin
+    {
+        s.governanceTokenAddress = _governanceTokenAddress;
+    }
+
+    function setSushiSwapPoolAddress(address _sushiSwapPoolAddress)
+        external
+        onlyAdmin
+    {
+        s.sushiSwapPoolAddress = _sushiSwapPoolAddress;
+    }
+
+    function setCreditCalculatorAddress(address _creditCalculatorAddress)
+        external
+        onlyAdmin
+    {
+        s.creditCalculatorAddress = _creditCalculatorAddress;
+    }
+
+    function setCreditNFTCalculatorAddress(address _creditNFTCalculatorAddress)
+        external
+        onlyAdmin
+    {
+        s.creditNFTCalculatorAddress = _creditNFTCalculatorAddress;
+    }
+
+    function setDollarMintCalculatorAddress(
+        address _dollarMintCalculatorAddress
+    ) external onlyAdmin {
+        s.dollarMintCalculatorAddress = _dollarMintCalculatorAddress;
```

Comment:
calculators are also a good fit for a facet

Reply:
> calculators are also a good fit for a facet

You are mentioning the possibility of implementing diamond on DollarMintCalculator.sol?
If yes, yeah, absolutely agree